rand_core/src/lib.rs

```diff
@@ -185,22 +209,51 @@ pub trait SeedableRng<Seed>: Rng {
     fn from_seed(seed: Seed) -> Self;
 }
 
+/// Error type for random number generators.
+#[derive(Copy, PartialEq, Eq, Clone, Debug)]
+pub enum Error {
+    /// This RNG is permanently unavailable.
+    NotAvailable,
+    /// Needs (re)seeding. Depending on the RNG this can mean you have to wait
+    /// and retry later, or you have to (re)seed it.
+    EntropyLow,
+    /// Failure, retry for a limited number of times.
+    TemporaryFailure,
+    /// Interrupted, usually by a signal from the OS.
+    Interrupted,
+    /// Any RNG error not part of this list.
+    Other,
+    #[doc(hidden)]
+    __Nonexhaustive,
+}
 
-/// Error type for cryptographic generators. Technically external generators
-/// such as the operating system or hardware generators could fail. A PRNG
-/// (algorithm) could also fail if it detects cycles, though most PRNGs have
-/// sufficiently long cycles that looping is not usually feasible.
-/// 
-/// TODO: how should error details be reported?
-#[derive(Debug)]
-pub struct Error;
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::NotAvailable => write!(f, "RNG permanently unavailable."),
+            Error::EntropyLow => write!(f, "RNG not seeded, or entropy exhausted."),
+            Error::TemporaryFailure => write!(f, "RNG has failed, probably temporary."),
+            Error::Interrupted => write!(f, "Rng operation interrupted."),
+            Error::Other => write!(f, "An unspecified RNG error occurred."),
+            Error::__Nonexhaustive => unreachable!(),
+        }
+    }
+}
 
 #[cfg(feature="std")]
-impl From<::std::io::Error> for Error {
-    fn from(_: ::std::io::Error) -> Error {
-        Error
+impl ::std::error::Error for Error {
+    fn description(&self) -> &str {
+        match *self {
+            Error::NotAvailable => "permanently unavailable",
+            Error::EntropyLow => "not seeded, or entropy exhausted",
+            Error::TemporaryFailure => "temporary failure",
+            Error::Interrupted => "operation interrupted",
+            Error::Other => "other rng error",
+            Error::__Nonexhaustive => unreachable!(),
+        }
     }
-}
 
-/// Result type (convenience type-def)
-pub type Result<T> = ::std::result::Result<T, Error>;
+    fn cause(&self) -> Option<&::std::error::Error> {
+        None
+    }
```

Comment:
I don't think there's any point implementing `cause` if it just returns `None`; this is the default impl.

Reply:
Didn't notice that. No, than I can happily remove it